backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/ProjectExportService.java

```diff
@@ -207,67 +206,65 @@ public ProjectExportService(IProjectService projectService, IDocumentService doc
-     * @throws IOException
-     *             if an I/O error occurred
+     * @return the {@link RepresentationManifest} for the given {@link RepresentationDescriptor}
      */
-    private RepresentationManifest createRepresentationManifest(Diagram diagram, RepresentationDescriptor representationDescriptor, ResourceSet resourceSet) throws IOException {
-        UUID descriptionId = diagram.getDescriptionId();
-        Optional<IdMappingEntity> optionalIdMappingEntity = this.idMappingRepository.findById(descriptionId);
-        if (optionalIdMappingEntity.isPresent()) {
-            IdMappingEntity idMappingEntity = optionalIdMappingEntity.get();
-
-            String uriFragment = ""; //$NON-NLS-1$
-            String targetObjectId = diagram.getTargetObjectId();
-            for (Resource resource : resourceSet.getResources()) {
-                EObject eObject = resource.getEObject(targetObjectId);
-                if (eObject != null) {
-                    uriFragment = EcoreUtil.getURI(eObject).toString();
-                    break;
-                }
-            }
-            if (uriFragment.isEmpty()) {
-                this.logger.error("The serialization of the representationManifest won't be complete."); //$NON-NLS-1$
-            }
-            // @formatter:off
-            return RepresentationManifest.newRepresentationManifest()
-                    .type(diagram.getClass().getSimpleName())
-                    .descriptionURI(idMappingEntity.getExternalId())
-                    .targetObjectURI(uriFragment)
-                    .build();
-            // @formatter:on
+    private RepresentationManifest createRepresentationManifest(RepresentationDescriptor representationDescriptor, ResourceSet resourceSet) {
+        IRepresentation representation = representationDescriptor.getRepresentation();
+        UUID descriptionId = representationDescriptor.getDescriptionId();
+
+        /*
+         * If the given descriptionId does not match with an existing IdMappingEntity, the current representation is
+         * based on a custom description. We use the descriptionId as descriptionURI.
+         */
+        // @formatter:off
+        String descriptionURI = this.idMappingRepository.findById(descriptionId)
+            .map(IdMappingEntity::getExternalId)
+            .orElse(descriptionId.toString());
+        // @formatter:on
 
+        String uriFragment = ""; //$NON-NLS-1$
+        String targetObjectId = representationDescriptor.getTargetObjectId();
+        for (Resource resource : resourceSet.getResources()) {
+            EObject eObject = resource.getEObject(targetObjectId);
+            if (eObject != null) {
+                uriFragment = EcoreUtil.getURI(eObject).toString();
+                break;
+            }
+        }
+        if (uriFragment.isEmpty()) {
+            this.logger.error("The serialization of the representationManifest won't be complete."); //$NON-NLS-1$
```

Comment:
I agree with you, I have opened an issue on this subject since it would be a bit complex to change this right now and it's not related to this commit. https://github.com/eclipse-sirius/sirius-components/issues/187